### ISSUE TITLE
Images are broken

### ISSUE THREAD

**Issue by tobiasredel** [Chuchodavids/EpGo#3]
SchedulesDirect recently introduced some changes in how images are downloaded (I think). At least i can not manage to get guide2go to include image links into the individual shows - they are all gone from my xml.
Any idea if this can be fixed as part of this guide2go fork?

Thank you in advance!!

**Comment by isaksamsten** [comment]
Hi!

I stumbled upon this issue and thought I could offer an alternative.

I've been using guide2go for some time, but it was never as feature-full as [tv_grab_az_sdjson_sqlite](https://github.com/azlm8t/tv_grab_az_sdjson_sqlite) which I stopped using when an authentication token was required for downloading icons.

When guide2go stopped downloading images, I took it upon myself to come up with a solution. My solution is a very simple python server that rewrites the `xmltv` generated by `tv_grab_az_sdjson_sqlite` and proxies (and caches) event icons.

I've been using it for about two weeks with some minor bugs (now fixed), and it seems pretty functional and stable. Check out the [sd_proxy repository](https://github.com/isaksamsten/sd_proxy) to see if it can help you too.

Cheers

**Comment by UpDryTwist** [comment]
Problem is that SD is sending down height and width as ints, not strings.  Straightforward fix.  Also added some logging (as long as you have "Show download errors ..." set in the config) to identify future issues.  I have some concerns that "size" might also be a problem, but don't have that in any feed I can test at this point.

Pull request is here:  #4  (chuchodavids will need to approve the PR to have it incorporated in the main build).

**Comment by isaksamsten** [comment]
Clone the repository and build the image.

    docker build . -t guide2go-dev

should produce an image `guide2go-dev` that you can use, until the official image is available.

**Comment by iawa2k** [comment]
> Clone the repository and build the image.
> 
> ```
> docker build . -t guide2go-dev
> ```
> 
> should produce an image `guide2go-dev` that you can use, until the official image is available.

You sir are a legend. Many thanks :)

**Comment by aschulenberg** [comment]
still hasn't been updated on the dockerhub

**Comment by iawa2k** [comment]
> Can anyone confirm this is working now?

I can, been working perfectly since I built the image in docker from source on the 4th Jan 2024.

**Comment by nilleiz** [comment]
@iawa2k 
Does your epg also include the NEW and LIVE icons now, i cant get rid of them when building with `docker build . -t guide2go-dev`.
Cloned the repository with `git clone https://github.com/Chuchodavids/guide2go.git .` is that correct?

**Comment by iawa2k** [comment]
> @iawa2k Does your epg also include the NEW and LIVE icons now, i cant get rid of them when building with `docker build . -t guide2go-dev`. Cloned the repository with `git clone https://github.com/Chuchodavids/guide2go.git .` is that correct?

Yes, I never disabled these as I like to see if it’s a new, live or old/prerecorded episode etc.

I can see you’ve opened a ticket so I’ll let any discussion continue there though. Images downloads are fixed though (the crux of this ticket issue), It’s just remaining open until the merge.

**Comment by tobiasredel** [comment]
any update on this topic? Can the fix be also implented on docker hub?

**Comment by Chuchodavids** [comment]
fixed
